Modifying Plugin Version in MANIFEST.MF displays ERROR log

### Describe the bug
Reproducible in 5.2.4 (latest)
Reported here: https://dotcms.zendesk.com/agent/tickets/100409
#### Important: This doesn't prevent the plugin to be deployed correctly

### Steps to reproduce the behavior:
1. deploy plugins
2. start dotCMS
3. stop dotCMS
4. undeploy plugins
5. change version in plugins/com.dotcms.config/MANIFEST.MF from 1.0 to 1.1​
6. deploy plugins
7. start dotCMS

check dotcms.log and see this error:
```
[06/02/20 09:03:01:553 CET] ERROR util.JDBCExceptionReporter: Batch entry 0 insert into plugin (plugin_name, plugin_version, author, first_deployed_date, last_deployed_date, id) values ('dotCMS Config', '1.1', 'dotCMS', '2010-06-17 00:00:00-04', '2020-02-06 03:03:01.532-05', 'com.dotcms.config') was aborted: ERROR: duplicate key value violates unique constraint "plugin_pkey"
[06/02/20 09:03:01:553 CET] ERROR util.JDBCExceptionReporter: ERROR: duplicate key value violates unique constraint "plugin_pkey"
[06/02/20 09:03:01:554 CET] ERROR util.JDBCExceptionReporter: Batch entry 0 insert into plugin (plugin_name, plugin_version, author, first_deployed_date, last_deployed_date, id) values ('dotCMS Config', '1.1', 'dotCMS', '2010-06-17 00:00:00-04', '2020-02-06 03:03:01.532-05', 'com.dotcms.config') was aborted: ERROR: duplicate key value violates unique constraint "plugin_pkey"
[06/02/20 09:03:01:554 CET] ERROR util.JDBCExceptionReporter: ERROR: duplicate key value violates unique constraint "plugin_pkey"
Caused by: java.sql.BatchUpdateException: Batch entry 0 insert into plugin (plugin_name, plugin_version, author, first_deployed_date, last_deployed_date, id) values ('dotCMS Config', '1.1', 'dotCMS', '2010-06-17 00:00:00-04', '2020-02-06 03:03:01.532-05', 'com.dotcms.config') was aborted: ERROR: duplicate key value violates unique constraint "plugin_pkey"
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "plugin_pkey"
```
### Expected behavior
No errors in the log.

--- a/dotCMS/src/main/java/com/dotmarketing/plugin/business/PluginFactoryDBImpl.java
+++ b/dotCMS/src/main/java/com/dotmarketing/plugin/business/PluginFactoryDBImpl.java
@@ -116,7 +116,17 @@ public class PluginFactoryDBImpl extends PluginFactory {
 	 */
 	@Override
 	protected void save(Plugin plugin) throws DotDataException {
-		HibernateUtil.saveOrUpdate(plugin);
+		final HibernateUtil hu = new HibernateUtil(Plugin.class);
+		final Plugin existingPlugin = (Plugin) hu.get(plugin.getId());
+		if (existingPlugin != null) {
+			existingPlugin.setPluginName(plugin.getPluginName());
+			existingPlugin.setPluginVersion(plugin.getPluginVersion());
+			existingPlugin.setAuthor(plugin.getAuthor());
+			existingPlugin.setLastDeployedDate(plugin.getLastDeployedDate());
+			HibernateUtil.update(existingPlugin);
+		} else {
+			HibernateUtil.save(plugin);
+		}
 		if(UtilMethods.isSet(plugin.getId())){
 			cache.removePlugin(plugin.getId());
 			cache.add(plugin);

